sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -412,15 +412,28 @@ case class DataSource(
     //  2. Output path must be a legal HDFS style file system path;
     //  3. It's OK that the output path doesn't exist yet;
     val allPaths = paths ++ caseInsensitiveOptions.get("path")
-    val outputPath = if (allPaths.length == 1) {
+    val (outputPath, pathExists) = if (allPaths.length == 1) {
       val path = new Path(allPaths.head)
       val fs = path.getFileSystem(sparkSession.sessionState.newHadoopConf())
-      path.makeQualified(fs.getUri, fs.getWorkingDirectory)
+      val qualifiedOutputPath = path.makeQualified(fs.getUri, fs.getWorkingDirectory)
+      (qualifiedOutputPath, fs.exists(qualifiedOutputPath))
     } else {
       throw new IllegalArgumentException("Expected exactly one path to be specified, but " +
         s"got: ${allPaths.mkString(", ")}")
     }
 
+    if (pathExists) {
```

Comment:
But in `InsertIntoHadoopFsRelationCommand` we still need to check path existence,  I do not quite agree with this refactor.

Anyway, let's revert it and put the refactor in a new PR, to unblock this bug fix patch